physionet-django/project/models.py

```diff
@@ -778,42 +864,56 @@ def citation_text(self, style):
         ----------
         style [string]:
             ['MLA', 'APA', 'Chicago', 'Harvard', 'Vancouver']
+        platform [string]:
+            the name of the platform / article publisher
+        authors [list]:
+            the authors of the article
+        year [str]:
+            the year the article was published
+        doi [str]:
+            the doi for the article
+        title [str]:
+            the title of the article
+        version [str]:
+            the version / volume of the article
 
         Returns
         -------
         citation_format [string]:
             string containing the desired citation style
         """
-        authors = self.authors.all().order_by('display_order')
+        if not authors:
+            authors = self.authors.all().order_by('display_order')
 
-        if self.is_published():
+        if not year and not doi:
+            if self.is_published():
 
-            year = self.publish_datetime.year
-            doi = self.doi
+                year = self.publish_datetime.year
+                doi = self.doi
 
-            if self.is_legacy:
-                return ''
+                if self.is_legacy:
+                    return ''
 
-        else:
-            """
-            Since the project is not yet published, the current year is used in
-            place of the publication year, and '*****' is used in place of the
-            DOI suffix.
-            """
-            year = timezone.now().year
-            doi = '10.13026/*****'
+            else:
+                """
```

Comment:
`"""` isn't Python syntax for a comment as far as I'm aware (though might be wrong!). Generally best to use `#` at the start of a line.

Reply:
That's used for a multi-line comment.. is there a PEP8 preference? I didn't show up in my linter

Reply:
It's a multiline string that hasn't been assigned to a variable, rather than a comment.

Reply:
Okay, I'll change it since I guess it's preferably used with docstrings, it was with the old code from whenever it was first developed